### DISCUSSION TITLE
ilogtail如何对某一个采集配置进行停止采集

### DISCUSSION THREAD

**Discussion by onejimmyboy** [alibaba/ilogtail#1433] [Help] [answered]
ilogtail如何对某一个采集配置进行停止采集
想要的效果是
1：如何对某一个采集配置停止采集后，再开启，采集日志不会重复采，继续在停止采集的位置继续采集

**Comment by onejimmyboy** [comment]
自问自答？

**Comment by messixukejia** [comment, marked as answer]
采集配置里有enable参数，可以控制停止。但是停止后就关闭采集了，再次开启会认为是全新配置，无法保持上次的位置。

可以描述下具体的需求场景，再评估下是否支持。

**Reply by onejimmyboy** [reply]
具体的需求就是：ilogtail输出到es中，es异常的情况下，需要主动停止采集，等es恢复了，这时候再开始采集，尽量保证采集日志不重复，不丢失

**Reply by messixukejia** [reply]
文件采集场景，如果下游异常，会无限重试，相当于停止采集了。ES Flusher可以测试下是否支持无限重试，如果不支持可以根据返回码做下判断。

**Reply by onejimmyboy** [reply]
这样的话ilogtail会一直在刷新错误，也会占用资源，影响正常采集的任务的性能。其实想要的是在ilogtail采集日志阶段就停止采集。

**Reply by messixukejia** [reply]
这里并不是一个简单关开的问题，停了之后还涉及hold句柄，会有一系列的问题需要处理。

该需求针对后续整体的反馈在做评估吧。

**Reply by onejimmyboy** [reply]
之前想的，像利用enable=false这个，如果是fasle，文件现在就不采集了，但是checkpoint会被删除掉。即使不被删除。在enable=true时，也不会从checkpoint中读取上次记录。如果想做到enable=fasle时，不删除checkpoint，同时enable=true从checkpoint读取就好了。
